sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -177,8 +180,7 @@ class NodeHttpClient implements HttpClient {
         if (isReadableStream(responseStream)) {
           downloadStreamDone = isStreamComplete(responseStream);
         }
-
-        Promise.all([uploadStreamDone, downloadStreamDone])
+        await Promise.all([uploadStreamDone, downloadStreamDone])
```

Comment:
I don't think we want to await this, since the promise that returns the response has to complete in order for the client to be able to finish reading the download stream, so either we would never complete the response or it gets ignored, I'm not sure how await in `finally` is handled

Reply:
I kinda agree that we should not await here especially when download streaming is a time-consuming one. I have a [playground ](https://www.typescriptlang.org/play/?ssl=20&ssc=7&pln=20&pc=10#code/IYZwngdgxgBAZgV2gFwJYHsIwE5IBQCUMA3gFAwUzLZgnmUPIAW26A7jBAKYcCi2rbHgDkXAemzCCAbnoUAvjCjBkUJjDxjsALhjAIYImQYMomEOgA2XAHSX0AcxFN9AE0uoIDmFonCAND4CNgC2XCAgwA5cBHIwinCewJaWtMYmFGYQFtZ2jiJQ1vqe3ggADlJxDMBswKjIMCDWXGV4AKwADB0yVZRZObb2TqIAHvUlUrIM8nHYXMgI2FjCrphcwrIzpIgoGFhNXC14ISC6EAghAEZiRvRzC0ucPDAACqwhqCBceHhzOQBuMRgAF4AHx0BhfZAAFVQYXQCGQv3CVkBgROPQUPS2-QafxBOHwPT+NmYXAgeH+yRB4P6VkG+SplgIBCAA) and so I guess the whole promise would be blocked until the promise in finally is completed and this may be not what we want. 

Back to the original issue the listener is added during polling and the removal will be triggered immediately because the responseStream is consumed to the end already and downloadStreamDone here should be in resolved/fulfilled state. Which means even without this await, once we have isStreamComplete [improvement](https://github.com/Azure/azure-sdk-for-js/pull/31027/files#r1750628330), these listeners would be removed correctly.

https://github.com/Azure/azure-sdk-for-js/blob/f405797940b98e91335447c8bfb282be744131d0/sdk/core/core-rest-pipeline/src/nodeHttpClient.ts#L161-L171